### PR TITLE
Changes to Networking and Yosemite layers for deleting an existing Product

### PR DESCRIPTION
**Pull request by pmusolino**
Yosemite/Networking changes for #2853 

## Changes
- Networking layer:
  - In `ProductsRemote`, added `deleteProduct` for deleting (trashing) a product + tests
- Yosemite layer:
  - Added `ProductAction.deleteProduct`
  - In `ProductStore`, implemented `deleteProduct` + tests
  - Updated `MockProductsRemote` for unit tests

## Testing
This PR doesn't affect existing behavior, just CI!

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/NetworkingTests/Remote/ProductsRemoteTests.swift [hunk 1]
```diff
@@ -125,6 +125,106 @@ final class ProductsRemoteTests: XCTestCase {
...
     }
 
+    // MARK: - Delete Product
+
+    func test_deleteProduct_with_success_mock_returns_a_product() {
+        // Given
+        let remote = ProductsRemote(network: network)
+
+        /// When we delete a product, it return the deleted product.
+        network.simulateResponse(requestUrlSuffix: "products/\(sampleProductID)", filename: "product-add")
```
Comment (jaclync) [inline Networking/NetworkingTests/Remote/ProductsRemoteTests.swift]: nit: if both adding and deleting a product returns the same response, maybe we can rename the `product-add.json` to include the delete case like `product-add-or-delete.json`?

File: Networking/NetworkingTests/Remote/ProductsRemoteTests.swift [hunk 2]
```diff
@@ -125,6 +125,106 @@ final class ProductsRemoteTests: XCTestCase {
...
+                                      images: [],
+                                      attributes: [],
+                                      defaultAttributes: [],
+                                      variations: [],
+                                      groupedProducts: [],
+                                      menuOrder: 0)
+        XCTAssertEqual(deletedProduct, expectedProduct)
+    }
+
+    func test_deleteProduct_relays_netwoking_error() {
```
Comment (jaclync) [inline Networking/NetworkingTests/Remote/ProductsRemoteTests.swift]:
```suggestion
    func test_deleteProduct_relays_networking_error() {
```

**Review by jaclync — APPROVED** [review]
Nice work! If you've tested deleting a product, do you know if it trashes the product or deletes the product permanently on the server? (from this discussion C6H8C3G23/p1600696854038900-slack)

File: Networking/NetworkingTests/Remote/ProductsRemoteTests.swift (hunk 1 above)
Comment (pmusolino) [reply Networking/NetworkingTests/Remote/ProductsRemoteTests.swift]: It makes sense 👍 . Changed here: 0de6a7a9b2e7060c5b8e7c1ad5cfc2ed07974b9c

**Comment by pmusolino** [comment]
Thank you Jaclyn for the review!

> Nice work! If you've tested deleting a product, do you know if it trashes the product or deletes the product permanently on the server? (from this discussion C6H8C3G23/p1600696854038900-slack)

Yes. The [API request](https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-a-product) by default move the product into the trash. If we want to delete it permanently, we should pass `force = true`. 🙌

**Pull request merged**